server/backend/feedback.go

```diff
@@ -0,0 +1,160 @@
+package backend
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	pb "github.com/TUM-Dev/Campus-Backend/server/api/tumdev"
+	"github.com/TUM-Dev/Campus-Backend/server/backend/cron"
+	"github.com/TUM-Dev/Campus-Backend/server/model"
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/gofrs/uuid/v5"
+	"github.com/guregu/null"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+// SendFeedback accepts a stream of feedback messages from the client and stores them in the database/file system.
+func (s *CampusServer) SendFeedback(stream pb.Campus_SendFeedbackServer) error {
+	return s.db.WithContext(stream.Context()).Transaction(func(tx *gorm.DB) error {
+		// receive metadata
+		req, err := stream.Recv()
+		if err == io.EOF {
+			return status.Error(codes.InvalidArgument, "no actual feedback provided") // The client has closed the stream
+		}
+		if err != nil {
+			log.WithError(err).Error("Error receiving feedback")
+			return status.Error(codes.Internal, "Error receiving feedback")
+		}
+		imageCounter := int32(0)
+		feedback, err := convertReqToFeedback(req)
+		if err != nil {
+			return status.Error(codes.Internal, "Error converting request to feedback")
+		}
+		if len(req.Attachment) > 0 {
+			imageCounter = handleImageUpload(tx, &req.Attachment, imageCounter, feedback.EmailId.String)
+		}
+
+		// download images
+		for {
+			req, err := stream.Recv()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				log.WithError(err).Error("Error receiving feedback")
+				deleteUploaded(tx, feedback.EmailId.String)
+				return status.Error(codes.Internal, "Error receiving feedback")
+			}
+
+			if len(req.Attachment) > 0 {
```

Comment:
This seems to be duplicated line number 39?

Reply:
I have changed the code from assuming that the first sent entry includes actual feedback to allow upding later on => removj g this concearn